sentinel-core/utils/file_utils.cc

```diff
@@ -0,0 +1,81 @@
+#include "sentinel-core/utils/file_utils.h"
+
+#include <dirent.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+
+#include <errno.h>
+#include <string.h>
+
+#include <fstream>
+#include <iostream>
+
+namespace Sentinel {
+namespace Utils {
+
+bool FileUtils::FileExists(const std::string& path) {
+  std::ifstream input_file(path);
+  return input_file.is_open();
+}
+
+bool FileUtils::DirExists(const std::string& path) {
+  DIR* const dir = ::opendir(path.c_str());
```

Comment:
If `opendir` returns `nullptr`, it does not mean that the directory does not exist, there may be a permission problem, the error code needs to be verified is ENOENT